spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -270,6 +278,23 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
     this.createOrcAsIceberg = options.getBoolean("orc-enabled", createOrcAsIceberg);
   }
 
+  private void validateHmsUri(String catalogHmsUri) {
+    if (catalogHmsUri == null) {
+      return;
+    }
+
+    Configuration conf = SparkSession.active().sessionState().newHadoopConf();
+    String envHmsUri = conf.get(HiveConf.ConfVars.METASTOREURIS.varname, null);
+    if (envHmsUri == null) {
+      return;
+    }
+
+    Preconditions.checkArgument(catalogHmsUri.equals(envHmsUri),
+        "The environment Hive metastore uri(%s) doesn't match with Spark session catalog uri(%s). " +
+            "Please make sure they are the same, or keep a valid one and unset the other.",
```

Comment:
We normally try to keep error messages short and only include tips for resolving problems if they aren't obvious. Here, I'd recommend: `"Inconsistent Hive metastore URIs: %s (Spark session) != %s (spark_catalog)", envHmsUri, catalogHmsUri`